### ISSUE TITLE
iOS VoiceOver doesn't announce state of details element when using role group

### ISSUE THREAD

**Issue by selfthinker** [alphagov/reported-bugs#28]
**Upstream bug:** https://bugs.webkit.org/show_bug.cgi?id=180865
**Date:** 2017-12-15
**Reported by:** Anika Henke
**Related to:** https://github.com/alphagov/govuk_elements/issues/573

---

### Overview

When you access a details element with VoiceOver in iOS 10 or iOS 11, it doesn't announce that it can be expanded when it also has the ARIA role "group".

Although the group role is the default implicit role of the details element [http://w3c.github.io/html/interactive-elements.html#elementdef-details] and it is recommended not to use it in such a case [http://w3c.github.io/html/dom.html#do-not-set], it should also not make the element behave any differently.

### Steps to reproduce

1. Activate VoiceOver
2. Open http://jsbin.com/cequcu in Safari
3. Navigate to the first details element ("Open this")

### Actual result

VoiceOver says "Open this".

### Expected result

VoiceOver should say "Open this, collapsed" like it does in the second example (which doesn't include the group role).

**Comment by selfthinker** [comment]
@36degrees wondered if this might have been fixed, at least in Safari Technology Preview? https://developer.apple.com/safari/technology-preview/release-notes/#r59

But when I retested in November 2018 in iOS 12.0.1 and iOS 12.1 it wasn't fixed in those yet.

It's be worth re-testing to check if this is still an issue or not.

**Comment by querkmachine** [comment]
The issue of it not announcing the text when receiving virtual focus appears to have been fixed since this bug was raised. On iOS 26.4, focusing over a `summary` element with the 'group' `role` announces this:

> Open this, button, collapsed. Double tap to expand. 

Moving focus away and then back to the `summary` also correctly announces: 

> Open this, button, expanded. Double tap to collapse. 

The announcements are identical for native `summary` elements without a `role` applied.  

VoiceOver **does** bug out when expanding or collapsing, however. It begins to announce what is happening—"expanding" or "collapsing", presumably—but is cut off very quickly, usually only managing to make "eh" or "cuh" sounds before stopping. This happens with all `details`/`summary` elements, however, and is not specific to having the 'group' `role`.

As the originally reported bug appears to have been rectified, I'm going to close this as fixed. 

The issue with VoiceOver not announcing state changes appears to already have a bug report open, which I've added a comment to: https://bugs.webkit.org/show_bug.cgi?id=263498
